lib/src/utils.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:io/io.dart';
+
+class CommonUtils {
+  ProcessManager _manager;
+
+  CommonUtils() {
+    this._manager = new ProcessManager();
+  }
+
+  /// Fetches the configlet file if it doesn't exist already, and returns the
+  /// exit code.
+  int fetchConfiglet() {
+    File configletFile = new File('bin/configlet');
+
+    if (!configletFile.existsSync()) {
+      print('Fetching configlet...');
+      return _exit(Process.runSync('bin/fetch-configlet', []).exitCode);
+    }
+
+    return 0;
+  }
+
+  /// Returns a [Future] with the exit code resulting from running the
+  /// [executable] with [arguments].
+  ///
+  /// If the [executable] isn't executable, returns exit code 1 and shows an
+  /// error message.
+  Future<int> runCmd(String executable, [List<String> arguments = const []]) async {
+    // Commented out because it always return false with "pub".
+    //
+    // final result = isExecutable(executable);
+    // if (result is bool && !result || result is Future && !await result) {
+    //   print('Unable to run "$executable". Make sure that it\'s executable and that you have permission to run it.');
+    //   return new Future.value(1);
+    // }
```

Comment:
If we are not using this, recommend removing the commented code

Reply:
I recommended implementing such, but it needs debugging. I'll see what I can do about that.

Reply:
@kabiir I think I know what's going on here. Anything under our `bin/` is executable, but all other commands aren't (that includes `pub` and all other system commands like `ls` and `pwd`). Looking again at the docs for `isExecutable()`, the first parameter is a path, and it returns `true` if this path contains a file that is executable. Well, commands like `pub` aren't a path, but rather an alias to a path, and I'm guessing that's why `isExecutable()` returns `false`. Since we have no way of checking if a command is a path or an executable, I think we should probably get rid of this check. I'd like to know what you think.

Reply:
Maybe keep it for the scripts, but use an alternative for commands like `pub` without it

Reply:
Ok, that sounds like a good idea. One function that has the `isExecutable()` check, and another that doesn't, and then it's up to the user which one they want to use. Thanks guys.